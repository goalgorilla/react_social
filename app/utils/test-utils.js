// test-utils.js
import React from 'react';
import PropTypes from 'prop-types';
import {render} from '@testing-library/react';
import {ThemeProvider} from 'styled-components';
import GlobalStyle from '../components/GlobalStyle';
import theme from '../components/themes/theme';

const AllTheProviders = ({children}) => {
  return (
    <ThemeProvider theme={theme}>
      <React.Fragment>
        {children}
        <GlobalStyle />
      </React.Fragment>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options});

AllTheProviders.propTypes = {
  children: PropTypes.node,
};

// re-export everything
export * from '@testing-library/react';

// override render method
export {customRender as render};
