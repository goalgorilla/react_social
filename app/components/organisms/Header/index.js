import React from "react";
import styled from "styled-components";
import Logo from "../../atoms/Logo";
import NavigationList from "../../molecules/NavigationList";
import AccountNavigation from "../../molecules/AccountNavigation";
import NavigationDropdown from "../../molecules/NavigationDropdown";
import { deviceMinWidth } from "../../../utils/device";

// The header of the website. Containing the navigation bars.
const Wrapper = styled.div`
  position: fixed;
  z-index: 900;
  top: 0;
  display: flex;
  box-shadow: 0 -1px 0 #e0e0e0, 0 0 2px rgba(0, 0, 0, 0),
    0 2px 4px rgba(0, 0, 0, 0.32);
  width: 100%;
  height: 50px;
  background: ${props => props.theme.color.brand.primary};
  user-select: none;

  @media ${deviceMinWidth.tablet} {
    height: 100px;
  }
`;

const NavRow = styled.div`
  display: flex;
  box-sizing: border-box;
  width: 100%;
  height: 50px;
  font-size: ${props => props.theme.font.size.small};
  font-weight: ${props => props.theme.font.weight.bold};
  color: ${props => props.theme.color.text.light};

  & > a {
    margin-left: auto;
    margin-right: 15px;
    padding: 0 12.5px 0 12.5px;
    line-height: 50px;
    text-align: center;
    text-decoration: none;
    color: ${props => props.theme.color.text.three};
  }

  &:first-of-type {
    display: none;
    color: ${props => props.theme.color.text.three};
    background: ${props => props.theme.color.background.secondary};
    padding-left: 0.9375rem;
  }

  &:first-of-type > div {
    display: flex;
    align-items: center;
  }

  &:first-of-type div > li > a {
    display: block;
    height: 100%;
    line-height: 50px;
    padding: 0 0.6125rem 0 0.6125rem;
  }

  &:first-of-type div > li > a > img {
    vertical-align: middle;
    margin-top: -4px;
  }

  &:first-of-type > div > a {
    display: flex;
    align-items: center;
  }

  &:first-of-type > div img:first-of-type {
    padding-right: 10px;
    opacity: 0.9;
  }

  @media ${deviceMinWidth.tablet} {
    &:first-of-type {
      display: flex;
      position: relative;
    }
  }
`;

const NavWrapper = styled.div`
  flex-direction: row;
  display: flex;
  width: 100%;

  @media ${deviceMinWidth.tablet} {
    flex-direction: column;
  }
`;

const Header = () => (
  <Wrapper>
    <Logo></Logo>
    <NavWrapper>
      <NavRow>
        <NavigationDropdown
          button={
            <React.Fragment>
              <img src="/static/translate.svg" width="20px" />
              English
              <img src="/static/dropdown.svg" width="20px" />
            </React.Fragment>
          }
          navRowOne={true}
        >
          <ul>
            <li>
              <a>English (current)</a>
            </li>
            <li>
              <a>Afrikaans</a>
            </li>
            <li>
              <a>Albanian</a>
            </li>
            <li>
              <a>Amharic</a>
            </li>
            <li>
              <a>Arabic</a>
            </li>
            <li>
              <a>Armenian</a>
            </li>
            <li>
              <a>Azerbaijani</a>
            </li>
          </ul>
        </NavigationDropdown>
        <a href="http://www.getopensocial.com" target="_blank">
          getopensocial.com
        </a>
      </NavRow>
      <NavRow>
        <NavigationList />
        <AccountNavigation />
      </NavRow>
    </NavWrapper>
  </Wrapper>
);

Header.defaultProps = {};

Header.propTypes = {};

export default Header;
