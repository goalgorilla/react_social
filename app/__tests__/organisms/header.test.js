import React from "react";
import { render, fireEvent, cleanup } from "../test-utils/test-utils";
import Header from "../../components/organisms/Header";

afterEach(cleanup);

describe("Header renders", () => {
  it("Shows Logo", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("logo")).not.toBeNull();
  });

  it("Shows navigation links", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("header-nav")).not.toBeNull();
  });

  it("Shows account navigation", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("account-nav")).not.toBeNull();
  });

  it("Shows dropdown when a dropdown link is clicked", () => {
    const { getByText, getByTestId } = render(<Header />);
    fireEvent.click(getByText("Explore"));
    expect(getByTestId("dropdown-menu")).not.toBeNull();
  });

  it("is 100% width at the top(0px) of the page", () => {
    render(<Header />);

    const headerClass = Header(false, null, "mockUser").type.styledComponentId;
    const MyHeaderRoots = document.getElementsByClassName(headerClass);
    const style = window.getComputedStyle(MyHeaderRoots[0]);

    expect(style.position).toBe("fixed");
    expect(style.top).toBe("0px");
    expect(style.width).toBe("100%");
  });
});

describe("With React Testing Library Snapshot", () => {
  it("Should match Snapshot", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});
