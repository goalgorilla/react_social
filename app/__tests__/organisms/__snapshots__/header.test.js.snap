// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="Header__Wrapper-sc-1f0tfo1-0 gOOKAB"
  >
    <a
      class="Logo__StyledA-e7xfjo-0 jGzaBK"
    >
      <img
        alt="[Community Name]"
        class="Logo__StyledImg-e7xfjo-1 dLzScd"
        data-testid="logo"
        src="/static/logo.svg"
      />
    </a>
    <div
      class="Header__NavWrapper-sc-1f0tfo1-2 jQRpyd"
    >
      <div
        class="Header__NavRow-sc-1f0tfo1-1 frERno"
      >
        <div>
          <div
            class="container NavigationDropdown__Wrapper-uiy1lv-0 iXuusX"
          >
            <li
              class="NavigationDropdown__NavButton-uiy1lv-2 fxbPMi"
            >
              <a>
                <img
                  src="/static/translate.svg"
                  width="20px"
                />
                EnglishÂ 
                <img
                  src="/static/dropdown.svg"
                  width="20px"
                />
              </a>
            </li>
          </div>
        </div>
        <a
          href="http://www.getopensocial.com"
          target="_blank"
        >
          getopensocial.com
        </a>
      </div>
      <div
        class="Header__NavRow-sc-1f0tfo1-1 frERno"
      >
        <ul
          class="NavigationList__NavBar-sc-16etxpz-1 cioncu"
          data-testid="header-nav"
        >
          <div
            class="NavigationList__MobileWrapper-sc-16etxpz-2 bDxsgl"
          >
            <div>
              <div
                class="container NavigationDropdown__Wrapper-uiy1lv-0 iXuusX"
              >
                <li
                  class="NavigationDropdown__NavButton-uiy1lv-2 gKjZse"
                >
                  <a>
                    <img
                      src="/static/translate.svg"
                      width="20px"
                    />
                  </a>
                </li>
              </div>
            </div>
            <div>
              <div
                class="container NavigationDropdown__Wrapper-uiy1lv-0 iXuusX"
              >
                <li
                  class="NavigationDropdown__NavButton-uiy1lv-2 gKjZse"
                >
                  <a>
                    <img
                      src="/static/hamburger.svg"
                      width="20px"
                    />
                  </a>
                </li>
              </div>
            </div>
          </div>
          <div
            class="NavigationList__DesktopWrapper-sc-16etxpz-3 buWtxT"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <div>
                <div
                  class="container NavigationDropdown__Wrapper-uiy1lv-0 iXuusX"
                >
                  <li
                    class="NavigationDropdown__NavButton-uiy1lv-2 gKjZse"
                  >
                    <a>
                      Explore
                    </a>
                  </li>
                </div>
              </div>
            </ul>
          </div>
          <div>
            <div
              class="container NavigationDropdown__Wrapper-uiy1lv-0 iXuusX"
            >
              <li
                class="NavigationDropdown__NavButton-uiy1lv-2 gKjZse"
              >
                <a>
                  <img
                    src="/static/search.svg"
                    width="24px"
                  />
                </a>
              </li>
            </div>
          </div>
        </ul>
        <ul
          class="AccountNavigation__AccountNavigationWrapper-gfi0qn-1 cylFQq"
          data-testid="account-nav"
        >
          <div
            class="AccountNavigation__MobileWrapper-gfi0qn-4 fBFadr"
          >
            <div>
              <div
                class="container NavigationDropdown__Wrapper-uiy1lv-0 iXuusX"
              >
                <li
                  class="NavigationDropdown__NavButton-uiy1lv-2 gKjZse"
                >
                  <a>
                    <img
                      src="/static/account.svg"
                      width="24px"
                    />
                  </a>
                </li>
              </div>
            </div>
          </div>
          <div
            class="AccountNavigation__DesktopWrapper-gfi0qn-5 jWMLfD"
          >
            <ul>
              <li>
                <a
                  href="/signup"
                >
                  Sign up
                </a>
              </li>
              <li>
                <a
                  href="/login"
                >
                  Log in
                </a>
              </li>
            </ul>
          </div>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
